cloud-companion/src/main/kotlin/org/codefreak/cloud/companion/ProcessManager.kt

```diff
@@ -0,0 +1,77 @@
+package org.codefreak.cloud.companion
+
+import com.pty4j.PtyProcessBuilder
+import java.io.OutputStream
+import java.util.UUID
+import kotlin.io.path.absolutePathString
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.core.io.buffer.DataBuffer
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import reactor.core.scheduler.Schedulers
+
+@Service
+class ProcessManager {
+  private val processMap: MutableMap<UUID, Process> = mutableMapOf()
+  private val outputStreamCache: MutableMap<UUID, Flux<DataBuffer>> = mutableMapOf()
+
+  @Autowired
+  private lateinit var fileService: FileService
+
+  fun createProcess(cmd: List<String>): UUID {
+    val uid = UUID.randomUUID()
+    processMap[uid] = generateProcess(cmd)
```

Comment:
What if that id already exists? Do we just overwrite the existing process?

Reply:
Hmm yes... but the probability of collision is extreeeemly low

Reply:
Yeah, but it still possible and thus a bug. I guess it would be quite a pain to find the problem if it happens someday 😅

Reply:
Fixed